### PR TITLE
Replace boost::program_options with nitro::options

### PR DESCRIPTION
**Pull request by bmario**
Finally, this replaces the last remaining part of Boost in lo2s using nitro::options. For now, it uses a feature branch of nitro.

### CODE REVIEW

**Comment by bmario** [comment]
The current output of `lo2s --help`:

```
usage: lo2s [-AVXaghqv] --list-clockids --list-events --list-tracepoints
            --list-knobs --instruction-sampling --standard-metrics --no-ip
            --disassemble --kernel [--clockid] [--count] [--event]
            [--metric-count] [--metric-frequency] [--metric-leader]
            [--mmap-pages] [--output-trace] [--perf-readout-interval] [--pid]
            [--readout-interval] [--metric-event] [--tracepoint]
            [--x86-adapt-knob] [COMMAND ...]

Lightweight Node-Level Performance Monitoring

  ./lo2s [options] [-a | -A] COMMAND
  ./lo2s [options] [-a | -A] -- COMMAND [args to command...]
  ./lo2s [options] [-a | -A] -p PID



Options:
  -h [ --help ]                         Show this help message.
  -V [ --version ]                      Print version information.
  -q [ --quiet ]                        Suppress output.
  -v [ --verbose ]                      Verbose output (specify multiple times
                                        to get increasingly more verbose
                                        output).
  -o [ --output-trace ] ARG [=lo2s_trace_{DATE}]
                                        Output trace directory. Can be set
                                        using the environment variable
                                        'LO2S_OUTPUT_TRACE'.
  -p [ --pid ] PID                      Attach to process of given PID.
  -m [ --mmap-pages ] PAGES [=16]       Number of pages to be used by internal
                                        buffers.
  -i [ --readout-interval ] MSEC [=100] Amount of time between readouts of
                                        interval based monitors, i.e.
                                        x86_adapt, x86_energy.
  -I [ --perf-readout-interval ] MSEC   Maximum amount of time between readouts
                                        of perf based monitors, i.e. sampling,
                                        metrics, tracepoints. If not provided,
                                        interval based readouts are disabled.
  -k [ --clockid ] CLOCKID [=monotonic-raw]
                                        Reference clock used as timestamp
                                        source.
  --list-clockids                       List all available clockids.
  --list-events                         List available metric and sampling
                                        events.
  --list-tracepoints                    List available kernel tracepoint
                                        events.
  --list-knobs                          List available x86_adapt CPU knobs.

System-monitoring mode options:
  -a [ --all-cpus ]                     Start in system-monitoring mode for all
                                        CPUs. Monitor as long as COMMAND is
                                        running or until PID exits.
  -A [ --all-cpus-sampling ]            System-monitoring mode with instruction
                                        sampling. Shorthand for "-a
                                        --instruction-sampling".

Sampling options:
  --[no-]instruction-sampling [=yes]    Enable instruction sampling.
  -e [ --event ] EVENT [=cpu-clock]     Interrupt source event for sampling.
  -c [ --count ] N [=11010113]          Sampling period (in number of events
                                        specified by -e).
  -g [ --call-graph ]                   Record call stack of instruction
                                        samples.
  --no-ip                               Do not record instruction pointers [NOT
                                        CURRENTLY SUPPORTED]
  --[no-]disassemble [=no]              Enable augmentation of samples with
                                        instructions (default if supported).
  --[no-]kernel [=no]                   Include events happening in kernel
                                        space (default).

perf metric options:
  -E [ --metric-event ] EVENT           Record metrics for this perf event.
  --standard-metrics                    Enable a set of default metrics.
  --metric-leader EVENT                 The leading metric event.
  --metric-count N                      Number of metric leader events to
                                        elapse before reading metric buffer.
                                        Has to be used in conjunction with
                                        --metric-leader
  --metric-frequency HZ [=10]           Number of metric buffer reads per
                                        second. Can not be used with
                                        --metric-leader

Kernel tracepoint options:
  -t [ --tracepoint ] TRACEPOINT        Enable global recording of a raw
                                        tracepoint event (usually requires
                                        root).

x86_adapt options:
  -x [ --x86-adapt-knob ] KNOB          Add x86_adapt knobs as recordings.
                                        Append #accumulated_last for semantics.

x86_energy options:
  -X [ --x86-energy ]                   Add x86_energy recordings.
```

**Review by cvonelm — APPROVED** [review]

**Comment by cvonelm** [comment]
Looks fine to me and the lo2s command line I've used for the measurements in my bachelor thesis still does what it should do, although it is not the most sophisticated lo2s command line

**Comment by tilsche** [comment]
As per our discussion the totally incomplete formatting goal:

```
usage: lo2s [-AVXaghqv] [--list-clockids] --list-events --list-tracepoints
            --list-knobs [--[no-]instruction-sampling] --standard-metrics [--[no-]ip]
            --disassemble --kernel [--clockid] [--count] [--event]
            [--metric-count] [--metric-frequency] [--metric-leader]
            [--mmap-pages] [-o <ARG> | --output-trace <ARG>] [--perf-readout-interval] [--pid]
            [--readout-interval] [--metric-event] [--tracepoint]
            [--x86-adapt-knob] [COMMAND ...]

Lightweight Node-Level Performance Monitoring

  ./lo2s [options] [-a | -A] COMMAND
  ./lo2s [options] [-a | -A] -- COMMAND [args to command...]
  ./lo2s [options] [-a | -A] -p PID



Options:
  -h, --help                            Show this help message.
  -V [ --version ]                      Print version information.
  -q [ --quiet ]                        Suppress output.
  -v [ --verbose ]                      Verbose output (specify multiple times
                                        to get increasingly more verbose
                                        output).
  -o, --output-trace ARG
                                        Output trace directory. Can be set
                                        using the environment variable
                                        'LO2S_OUTPUT_TRACE' (default lo2s_trace_{DATE}).
  -p [ --pid ] PID                      Attach to process of given PID.
  -m [ --mmap-pages ] PAGES             Number of pages to be used by internal
                                        buffers. (default 16)
  -i [ --readout-interval ] MSEC        Amount of time in milliseconds between readouts of
                                        interval based monitors, i.e.
                                        x86_adapt, x86_energy. (default 100)
  -I [ --perf-readout-interval ] MSEC   Maximum amount of time between readouts
                                        of perf based monitors, i.e. sampling,
                                        metrics, tracepoints. If not provided,
                                        interval based readouts are disabled.

Sampling options:
  --[no-]instruction-sampling           Enable instruction sampling. (default enabled)
  
  
lo2s -i 100 (documented)
lo2s -i=100 (works but not documented)
lo2s --readout-interval 100 (documented)
lo2s --readout-interval=100 (works but not documented
```

**Comment by bmario** [comment]
The updated help message:

```
```

File: .gitmodules [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	url = https://github.com/tud-zih-energy/otf2xx.git
 [submodule "lib/nitro"]
 	path = lib/nitro
-	url = https://github.com/tud-zih-energy/nitro.git
+    url = https://github.com/tud-zih-energy/nitro.git
```
Comment (cvonelm) [inline .gitmodules:6]: bikeshedding: .gitmodules uses Tabs

File: src/trace/trace.cpp [hunk 2]
```diff
@@ -56,16 +56,6 @@ constexpr pid_t Trace::METRIC_PID;
...
-    if (prefix.size() == 0)
-    {
-        prefix = nitro::env::get("LO2S_OUTPUT_TRACE");
-    }
-
-    if (prefix.size() == 0)
-    {
-        prefix = "lo2s_trace_{DATE}";
-    }
-
```
Comment (cvonelm) [inline src/trace/trace.cpp:68]: Is it correct that this functionality should be removed? I don't see code that replaces this

**Review by cvonelm — CHANGES_REQUESTED** [review]

File: src/trace/trace.cpp (hunk 2 above)
Comment (bmario) [reply src/trace/trace.cpp:68]: There is:
https://github.com/tud-zih-energy/lo2s/blob/issue-147-prt4-broken-options/src/config.cpp#L137

**Review by cvonelm — APPROVED** [review]

**Comment by tilsche** [comment]
We can finally remove boost from `README.md`

:tada: :balloon: :partying_face: :tada: :balloon: :partying_face: :tada: :balloon: :partying_face:

**Pull request merged**